core-java-8/src/main/java/com/baeldung/bifunction/Calculator.java

```diff
@@ -0,0 +1,28 @@
+package com.baeldung.bifunction;
+
+import java.util.function.BiFunction;
+
+public class Calculator {
+	
+    private Integer a = 0;
+    private Integer b = 0;

```

Comment:
While I appreciate that the functional operations need `Integer`, I think this calculating pojo would naturally represent its own inner values using `int`. There's no meaning to `null` values as far as the calculator is concerned. This would affect the members and the constructor. The functional part would still use `Integer` and the compiler would do the necessary boxing/unboxing.